modules/xfem/src/userobjects/MeshCut3DUserObject.C

```diff
@@ -0,0 +1,1106 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "MeshCut3DUserObject.h"
+
+#include "XFEMFuncs.h"
+#include "MooseError.h"
+#include "libmesh/string_to_enum.h"
+#include "MooseMesh.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/edge_edge2.h"
+
+template <>
+InputParameters
+validParams<MeshCut3DUserObject>()
+{
+  InputParameters params = validParams<GeometricCutUserObject>();
+  params.addRequiredParam<std::string>("mesh_file", "Mesh file for XFEM geometric cuts");
+  params.addParam<Real>("size_control", "Criterion for refining elements while growing the crack");
+  params.addParam<unsigned int>("n_step_growth", "Number of steps for crack growth");
+  params.addClassDescription("Creates a UserObject for a mesh cutter in 3D problems");
+  return params;
+}
+
+// currently, this code does not allow predefined crack growth
+// all inital cracks have to be defined at t_start = t_end = 0
+MeshCut3DUserObject::MeshCut3DUserObject(const InputParameters & parameters)
+  : GeometricCutUserObject(parameters), _mesh(_subproblem.mesh())
+{
+  if (isParamValid("n_step_growth"))
+    _n_step_growth = getParam<unsigned int>("n_step_growth");
+  else
+    _n_step_growth = 0;
+
+  if (_n_step_growth == 0)
+    _grow = 0;
+  else
+    _grow = 1;
```

Comment:
No need to do it like this, you can supply default parameters when you define them in `validParams`.